pkg/provider/gitlab/gitlab.go

```diff
@@ -226,6 +246,31 @@ func (g *Gitlab) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretD
 	return secretData, nil
 }
 
+func matchesFilter(checkName bool, ref esv1beta1.ExternalSecretFind, environment string, data *gitlab.ProjectVariable) (bool, string) {
+	if environment != "" && environment != "*" {
+		// as of now gitlab does not support filtering of EnvironmentScope through the api call
+		if data.EnvironmentScope != environment {
+			return false, ""
+		}
+	}
+
+	secretName := data.Key
+	if data.Key == "" {
+		return false, ""
+	}
+
+	if checkName && !matchName(ref, secretName) {
+		return false, ""
+	}
+
+	return true, secretName
+}
+
+func matchName(ref esv1beta1.ExternalSecretFind, secretName string) bool {
+	matches, _ := regexp.MatchString(ref.Name.RegExp, secretName)
+	return matches
+}
```

Comment:
we should use `MatchName` method from `find` package to have a single source of truth on how regexp are handled.

Reply:
done